odoo/addons/base/models/res_lang.py

```diff
@@ -44,6 +47,12 @@ class LangDataDict(ReadonlyDict):
     """
     __slots__ = ()
 
+    def __getitem__(self, key: K) -> T:
+        try:
+            return self.__data[key]
+        except KeyError:
+            return self.__missing__(key)
```

Comment:
`key` is not used in `__missing__`. Sure if it is necessary to strictly follow the `dict.__missing__` API of python for this Proxy :thinking:

Reply:
> Sure if it is necessary to strictly follow the dict.__missing__ API of python for this Proxy
No, as we can choose the signature of the `__missing__` override. However, I think it is good practice to follow the Python API (and therefore pass the key)

Reply:
I wanted to keep the current `__missing__` method, but it is better to handle it directly in the exception manager (since the ReadonlyDict no longer triggers the call to `__missing__`).
I have made the change.